xla/service/hlo_computation_deduplicator.cc

```diff
@@ -83,10 +83,12 @@ StatusOr<bool> HloComputationDeduplicator::Run(
     // Ignore entry computation since it is called from outside and computations
     // with large number of instructions or large-size constants due to increase
     // in time taken to stringify.
+    // Also ignore computations that are associated with collective instructions.
     if (comp->IsEntryComputation() || comp->instruction_count() > 128 ||
-        ContainsLargeConstants(comp)) {
+        ContainsLargeConstants(comp) || comp->IsCollectiveCalledComputation()) {
```

Comment:
Can you please also add a test for this change?

Reply:
sorry this was supposed to be a test pr to trigger the CI, i will convert it to a draft for now